coda-prototype/deploy.sh

```diff
@@ -1,31 +1,29 @@
 #!/bin/bash
-set -e;
-
-# Delete last attempt if it exists
-rm -rf coda;
-
-# Create a temporary directory to export to
-mkdir coda;
-
-# Check out HEAD to the temporary directory.
-# Export via git archive, which exports as a zip which we have to unzip.
-git archive HEAD --output=coda/export.zip;
-unzip -qq coda/export.zip -d coda/;
-rm coda/export.zip;
-
-# Build project
-# Note: will need to run npm install here in future.
-tsc;
-
-# Get the commit hash and time-stamp
-hash=$(git rev-parse HEAD);
-date=$(git show -s --format=%ci HEAD)
-
-# Write the hash and time-stamp to a file
-echo "{\"hash\": \"$hash\", \"date\": \"$date\"}" >coda/version.json;
-
-# Zip up the output
-zip -r -qq coda.zip coda/;
-
-# Clean-up the temporary export directory.
-rm -rf coda;
+set -e
+
+# Revert version.json on exit.
+function finish {
+    echo $'{\n  "hash": "develop",\n  "date": ""\n}' >version.json
+}
+trap finish EXIT
+
+# If the current directory does not match the HEAD commit, emit a warning,
+# otherwise export the hash and date of HEAD to version.json
+changes=$(git status --porcelain) # Machine readable list of changes; empty if there no changes.
+if [ ! -z "${changes}" ]; then
+    echo "Warning: File system is different to HEAD, so the zipped project will not
+         include a version identifier."
+else
+    # Get the commit hash and time-stamp
+    hash=$(git rev-parse HEAD)
+    date=$(git show -s --format=%ci HEAD)
+
+    # Write the hash and time-stamp to a file
+    echo "{\"hash\": \"$hash\", \"date\": \"$date\"}" >version.json
+fi
+
+# Delete the previously deployed version, if it exists, so that it is not included in the next zip file.
+[ -e coda.zip ] && rm coda.zip
+
+# Zip this directory into coda.zip
+zip -r -qq coda.zip .
```

Comment:
Could you stick the version number on the end of the zip file? That will make it easier to upload to GitHub releases which we use to send the code see: https://github.com/lukechurch/coda/releases